packages/nextclade/src/tree/findPrivateMutations.cpp

```diff
@@ -317,8 +332,26 @@ namespace Nextclade {
       const auto aaDeletions =
         filter(seq.aaDeletions, [&geneName](const AminoacidDeletion& aaDel) { return aaDel.gene == geneName; });
 
-      result[geneName] =
+      const auto unknownAaRanges =
+        filter(seq.unknownAaRanges, [&geneName](const GeneAminoacidRange& unk) { return unk.geneName == geneName; });
+
+      auto resultForGene =
         findPrivateMutations<Aminoacid>(nodeMutMapForGene, seq, aaSubstitutions, aaDeletions, refPeptide);
+
+      // Filter out substitutions and deletions from results, where positions fall into ranges with
+      // aminoacid X in query peptide.
+      // In these cases the information is simply missing from the query and in the `processNodeMutations()` above
+      // if a substitution or a deletion is missing from query, but present in the node, a reversion is added to the
+      // results (a mutation from the node all the way to what was in the ref seq). We chose to remove these,
+      // because we believe that a reversion is a less likely event and most of these are the sequencing errors
+      // followed by uncertainty in translation.
+      resultForGene.privateSubstitutions = filter(resultForGene.privateSubstitutions,
+        [&unknownAaRanges](const AminoacidSubstitutionSimple& sub) { return !isUnknownAa(sub.pos, unknownAaRanges); });
+
+      resultForGene.privateDeletions = filter(resultForGene.privateDeletions,
+        [&unknownAaRanges](const AminoacidDeletionSimple& del) { return !isUnknownAa(del.pos, unknownAaRanges); });
+
```

Comment:
The filtering calls are here and the implementation for `isUnknownAa()` is at the top of the same file.